src/Chapter3.hs

```diff
@@ -479,7 +504,38 @@ After defining the city, implement the following functions:
    complicated task, walls can be built only if the city has a castle
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
-
+data Castle = Castle String | NoneCastle
+data Building = Church | Library
+data House = One | Two | Three | Four
+data Wall = Wall | NoWall
+data City = City{
+  cityCastle :: Castle,
+  cityWall :: Wall,
```

Comment:
This is the correct data type, that can represent what we have in requirements 👍🏼 
However, this also allows having some states that should not ideally exist: like having walls without any castle at all. With this representation, this is possible. However, we actually can restrict this on type-level by providing more representative types 🙂 This is actually the greatest power of Haskell and type system 😍 

This pattern is called "Make illegal states unrepresentable", and you can read more about it here:

* https://kowainik.github.io/posts/haskell-mini-patterns#make-illegal-states-unrepresentable